gsy_e_sdk/setups/Asset_API_scripts/rest_bid_offer_posting.py

```diff
@@ -0,0 +1,147 @@
+"""
+Template file for a trading strategy through the gsy-e-sdk api client using Rest.
+"""
+
+import os
+from time import sleep
+from gsy_e_sdk.aggregator import Aggregator
+from gsy_e_sdk.clients.rest_asset_client import RestAssetClient
+from gsy_e_sdk.utils import get_area_uuid_from_area_name_and_collaboration_id
+
+current_dir = os.path.dirname(__file__)
+ORACLE_NAME = "oracle"
+
+LOAD_NAMES = ["Load 1 L13", "Load 2 L21", "Load 3 L17"]
+PV_NAMES = ["PV 1 (4kW)", "PV 3 (5kW)"]
+STORAGE_NAMES = ["Tesla Powerwall 3"]
+AUTOMATIC = True
+
+
+class Oracle(Aggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.post_bid_offer()
+
+    def on_tick(self, tick_info):
+        pass
+
+    def post_bid_offer(self):
+        """Post a bid or an offer to the exchange."""
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            asset_info = area_dict.get("asset_info")
+            if not asset_info:
+                continue
+
+            # Consumption assets
+            required_energy = asset_info.get("energy_requirement_kWh")
+            if required_energy:
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=required_energy
+                )
+
+            # Generation assets
+            available_energy = asset_info.get("available_energy_kWh")
+            if available_energy:
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=available_energy
+                )
+
+            # Storage assets
+            buy_energy = asset_info.get("energy_to_buy")
+            if buy_energy:
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=buy_energy
+                )
+
+            sell_energy = asset_info.get("energy_to_sell")
+            if sell_energy:
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=sell_energy
+                )
+
+            self.execute_batch_commands()
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+def get_assets_name(node: dict) -> dict:
+    """
+    Parse the grid tree and return all registered assets
+    wrapper for _get_assets_name
+    """
+    if node == {}:
+        return {}
+    reg_assets = {"Area": [], "Load": [], "PV": [], "Storage": []}
+    _get_assets_name(node, reg_assets)
+    return reg_assets
+
+
+def _get_assets_name(node: dict, reg_assets: dict):
```

Comment:
We have a soft rule to use the type hints from the `typing` module (for Lists, Dictionaries, Sets:
```python
from typing import List, Dict
```
```suggestion
def _get_assets_name(node: Dict, reg_assets: Dict):
```